src/test/HostActivationTests/FrameworkResolution/RollForwardMultipleFrameworks.cs

```diff
@@ -0,0 +1,576 @@
+        [InlineData("6.1.1-preview.0", null,                                     false,                 "6.2.1")]
+        [InlineData("6.1.1-preview.0", null,                                     true,                  "6.1.1-preview.2")]
+        [InlineData("6.1.1-preview.0", Constants.RollForwardSetting.LatestPatch, false,                 "6.1.1-preview.2")]
+        [InlineData("6.1.1-preview.0", Constants.RollForwardSetting.Disable,     false,                 "[not found]")]
+        [InlineData("6.1.1-preview.2", Constants.RollForwardSetting.Disable,     false,                 "6.1.1-preview.2")]
+        public void ReconcileFrameworkReferences_InnerFrameworkReference_Release(
+            string versionReference,
+            string rollForward,
+            bool rollForwardToPreRelease,
+            string resolvedFramework)
+        {
+            CommandResult result = RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(MicrosoftNETCoreApp, "6.1.0")
+                    .WithFramework(MiddleWare, "2.1.0"),
+                dotnetCustomizer => dotnetCustomizer.Framework(MiddleWare).RuntimeConfig(runtimeConfig =>
+                    runtimeConfig.GetFramework(MicrosoftNETCoreApp)
+                        .WithRollForward(rollForward)
+                        .Version = versionReference),
+                rollForwardToPreRelease);
+
+            if (resolvedFramework == "[not found]")
+            {
+                result.ShouldFailToFindCompatibleFrameworkVersion();
+            }
+            else
+            {
+                result.ShouldHaveResolvedFrameworkOrFailedToReconcileFrameworkReference(
+                    MicrosoftNETCoreApp, resolvedFramework, versionReference, "6.1.0");
+            }
+        }
+
+        // Verify that inner framework reference 5.1.1 <defaults>
+        // is correctly reconciled with app's framework reference (<fxRefVersion>, <rollForward>).
+        // App fx reference is lower.
+        [Theory] // fxRefVersion  rollForward                               resolvedFramework
+        [InlineData("5.0.0",      Constants.RollForwardSetting.Disable,     null)]
+        [InlineData("5.1.3",      Constants.RollForwardSetting.Disable,     "5.1.3")]
+        [InlineData("5.0.0",      Constants.RollForwardSetting.LatestPatch, null)]
+        [InlineData("5.1.0",      Constants.RollForwardSetting.LatestPatch, "5.1.3")]
+        [InlineData("5.0.0",      null,                                     "5.1.3")]
+        [InlineData("5.0.0",      Constants.RollForwardSetting.Minor,       "5.1.3")]
+        [InlineData("5.0.0",      Constants.RollForwardSetting.LatestMinor, "5.1.3")]
+        [InlineData("1.0.0",      Constants.RollForwardSetting.Minor,       null)]
+        [InlineData("1.0.0",      Constants.RollForwardSetting.Major,       "5.1.3")]
+        [InlineData("1.0.0",      Constants.RollForwardSetting.LatestMajor, "5.1.3")]
```

Comment:
Add cases
`5.1.1`
`5.1.3-preview`

Reply:
Added `5.1.1`, the pre-release cases are covered in the pre-release methods (Added that case).